### PR TITLE
IG-2108 Get publisher teams endpoint

### PR DESCRIPTION
**Pull request by CiaraWardPA**
[IG-2108](https://hdruk.atlassian.net/browse/IG-2108)

- API to return required fields for all publisher teams
- Only return teams if logged in user is in the HDR admin team

### CODE REVIEW

File: src/resources/team/team.controller.js [hunk 1]
```diff
@@ -529,6 +529,46 @@ const deleteTeamMember = async (req, res) => {
...
+ * GET api/v1/teams/getList
+ *
+ * @desc Get the list of all publisher teams
+ *
+ */
+const getTeamsList = async (req, res) => {
+    try {
+        // 1. Get the publisher teams from the database 		
+		const teams = await TeamModel.find(
+			{ $and: [{ type: 'publisher' }, { active: true }]},
```
Comment (RobinKavPA) [inline src/resources/team/team.controller.js:542]: Overall looks really good Ciara, just a point of interest on this line - don't forget if you're only using 'AND' conditions, the filter query can be written as { type: 'publisher', active: true } , you only need to use the $and operator if you're combining with an OR.

File: src/resources/team/team.route.js [hunk 2]
```diff
@@ -5,13 +5,18 @@ import teamController from './team.controller';
 
 const router = express.Router();
 
+// @route   GET api/v1/teams/getList
+// @desc     Returns List of all Teams
+// @access   Private
+router.get('/getList', passport.authenticate('jwt'), teamController.getTeamsList);
```
Comment (RobinKavPA) [inline src/resources/team/team.route.js:11]: Only other thing I might change would be to make this is the GET all endpoint at '/' and then possibly pass the type parameter (publisher) as a query string

File: src/resources/team/team.controller.js (hunk 1 above)
Comment (CiaraWardPA) [reply src/resources/team/team.controller.js]: I've removed the $and

File: src/resources/team/team.route.js (hunk 2 above)
Comment (CiaraWardPA) [reply src/resources/team/team.route.js]: I've removed the 'getList' so it's just '/' now

**Review by PaulMcCaffertyPA — APPROVED** [review]

**Review by LewisKearseyPA — APPROVED** [review]

**Pull request merged**